redis.go

```diff
@@ -9,58 +9,85 @@ import (
 	rediscache "github.com/go-redis/redis/v8"
 )
 
+var (
+	_ Cache    = (*redisCache)(nil)
+	_ CacheCtx = (*redisCache)(nil)
+)
+
 type redisCache struct {
 	ctx    context.Context
 	client *rediscache.Client
 	ttl    time.Duration
-	Cache
 }
 
-func NewRedisCache(ctx context.Context, cacheURL string, ttl time.Duration) (Cache, error) {
+func NewRedisCacheWithClient(client *rediscache.Client, ttl time.Duration) *redisCache {
```

Comment:
by injecting the client the library user can customize it and build it any way its needed,

for example changing the database